src/Ocelot/ServiceDiscovery/Providers/PolingConsulServiceDiscoveryProvider.cs

```diff
@@ -15,42 +15,109 @@ namespace Ocelot.ServiceDiscovery.Providers
 {
     public class PollingConsulServiceDiscoveryProvider : IServiceDiscoveryProvider
     {
+        private readonly ConsulRegistryConfiguration _config;
         private readonly IOcelotLogger _logger;
-        private readonly IServiceDiscoveryProvider _consulServiceDiscoveryProvider;
-        private readonly Timer _timer;
-        private bool _polling;
+        private readonly IConsulClient _consul;
+        private const string VersionPrefix = "version-";
         private List<Service> _services;
-        private string _keyOfServiceInConsul;
 
-        public PollingConsulServiceDiscoveryProvider(int pollingInterval, string keyOfServiceInConsul, IOcelotLoggerFactory factory, IServiceDiscoveryProvider consulServiceDiscoveryProvider)
-        {;
+        private int _pollingInterval;
+        private ulong _waitIndex;
+
+        public PollingConsulServiceDiscoveryProvider(int pollingInterval, ConsulRegistryConfiguration config, IOcelotLoggerFactory factory, IConsulClientFactory clientFactory)
+        {
+            ;
+            _pollingInterval = pollingInterval == 0 ? 10000 : pollingInterval;
             _logger = factory.CreateLogger<PollingConsulServiceDiscoveryProvider>();
-            _keyOfServiceInConsul = keyOfServiceInConsul;
-            _consulServiceDiscoveryProvider = consulServiceDiscoveryProvider;
-            _services = new List<Service>();
-        
-            _timer = new Timer(async x =>
+
+            _config = config;
+            _consul = clientFactory.Get(_config);
+
+            Task.Factory.StartNew(async () =>
```

Comment:
What happens if Poll() throws a transient exception e.g. Consul is down for a few seconds when it makes it's request? Will it start again? I think timer might be better if it starts the process again. However I don't know how this would work at the moment because the timer doesnt try catch or anything!

Reply:
>When this is provided, the HTTP request will "hang" until a change in the system occurs, or the maximum timeout is reached. A critical note is that the return of a blocking request is no guarantee of a change. It is possible that the timeout was reached or that there was an idempotent write that does not affect the result of the query.

when change service config, client should update at once. but use timer, there have some interva of two consecutive modificationsl. if i use try..catch.. in Poll(), would it be better?